larray/core.py

```diff
@@ -3762,6 +3801,63 @@ def split_axis(self, axis, sep='_', names=None, regex=None):
         split_axes = [Axis(axis_labels, name) for axis_labels, name in zip(axes_labels, names)]
         return self[:axis_index] + split_axes + self[axis_index + 1:]
 
+    def align(self, other, join='outer', axes=None):
+        """Align this axis collection with another.
+
+        This ensures all common axes are compatible.
+
+        Parameters
+        ----------
+        other : AxisCollection
+        join : {'outer', 'inner', 'left', 'right'}, optional 
+            Defaults to 'outer'.
+        axes : AxisReference or sequence of them, optional
+            Axes to align. Need to be valid in both arrays. Defaults to None (all common axes).
+
+        Returns
+        -------
+        (left, right) : (AxisCollection, AxisCollection)
+            Aligned collections
+
+        Examples
+        --------
+
+        >>> col1 = AxisCollection("a=a0..a1;b=b0..b2")
+        >>> col1
+        AxisCollection([
+            Axis(['a0', 'a1'], 'a'),
+            Axis(['b0', 'b1', 'b2'], 'b')
+        ])
+        >>> col2 = AxisCollection("a=a0..a2;c=c0..c0;b=b0..b1")
+        >>> col2
+        AxisCollection([
+            Axis(['a0', 'a1', 'a2'], 'a'),
+            Axis(['c0'], 'c'),
+            Axis(['b0', 'b1'], 'b')
+        ])
+        >>> aligned1, aligned2 = col1.align(col2)
+        >>> aligned1
+        AxisCollection([
+            Axis(['a0', 'a1', 'a2'], 'a'),
+            Axis(['b0', 'b1', 'b2'], 'b')
+        ])
+        >>> aligned2
+        AxisCollection([
+            Axis(['a0', 'a1', 'a2'], 'a'),
+            Axis(['b0', 'b1', 'b2'], 'b')
+        ])
+        """
+        assert join in {'outer', 'inner', 'left', 'right'}
+        other = other if isinstance(other, AxisCollection) else AxisCollection(other)
+        join_axes = list(OrderedSet(self.names) & OrderedSet(other.names)) if axes is None else axes
```

Comment:
Not yet. I will certainly break at this point. The way to handle them is relatively easy when there are only anonymous axes (ie do it by axis position) but when mixing anonymous and non-anonymous, this gets tricky and I don't really know what we should do.